management/server/sql_store.go

```diff
@@ -311,6 +311,32 @@ func (s *SqlStore) SavePeerLocation(accountID string, peerWithLocation *nbpeer.P
 	return nil
 }
 
+// SaveUsers saves the given list of users to the database.
+// It updates existing users if a conflict occurs.
+func (s *SqlStore) SaveUsers(account *Account) error {
+	for id, user := range account.Users {
+		user.Id = id
+		user.AccountID = account.Id
+		for id, pat := range user.PATs {
+			pat.ID = id
+			user.PATsG = append(user.PATsG, *pat)
+		}
+		account.UsersG = append(account.UsersG, *user)
+	}
+	return s.db.Clauses(clause.OnConflict{UpdateAll: true}).Create(&account.UsersG).Error
+}
+
+// SaveGroups saves the given list of groups to the database.
+// It updates existing groups if a conflict occurs.
+func (s *SqlStore) SaveGroups(account *Account) error {
```

Comment:
I think it is not memory efficient. If I understand well the reason of the "G" postfix it is just used by the Gorm. So after that, we always need to clean up because the garbage collector will not do it. Nevertheless, we should not mix the layers. The account structure holds the account info and the SQL-related things. We should create separate struct definition just for the SQL. It is more code but at the end, it will be safer. 

Now I recommend to:
- change the parameters of the function **SaveGroups(accountID string, groups  map[string]*nbgroup.Group)**
- create a new slice []nbgroup.Group  
- copy the relevant data to there
- change the db.clauses to use the new slice

So after it the GroupsG will be empty again

Reply:
Maybe the "**&gorm.Session{FullSaveAssociations: true}**" keyword help to you